File: packages/go/domain/subscription.go
package domain

import (
	"time"
)

type SubscriptionStatus string

const (
	SubscriptionStatusActive         SubscriptionStatus = "active"
	SubscriptionStatusExpired        SubscriptionStatus = "expired"
	SubscriptionStatusCancelled      SubscriptionStatus = "cancelled"
	SubscriptionStatusPendingPayment SubscriptionStatus = "pending_payment"
)

type Plan struct {
	ID           string                 `json:"id"`
	Name         string                 `json:"name"`
	Code         string                 `json:"code"`
	PriceWch     float64                `json:"price_wch"`
	MaxLiveBots  int                    `json:"max_live_bots"`
	MaxPaperBots int                    `json:"max_paper_bots"`
	Features     JSONObject             `json:"features"`
	CreatedAt    time.Time              `json:"created_at"`
}

type Subscription struct {
	ID            string             `json:"id"`
	UserID        string             `json:"user_id"`
	PlanID        string             `json:"plan_id"`
	Status        SubscriptionStatus `json:"status"`
	PaidAmountWch float64            `json:"paid_amount_wch"`
	StartedAt     time.Time          `json:"started_at"`
	ExpiresAt     *time.Time         `json:"expires_at,omitempty"`
	CreatedAt     time.Time          `json:"created_at"`
}

File: packages/shared-types/subscription.ts
import type { IsoDatetimeString, JsonObject } from './json';

export enum SubscriptionStatus {
  ACTIVE = 'active',
  EXPIRED = 'expired',
  CANCELLED = 'cancelled',
  PENDING_PAYMENT = 'pending_payment',
}

export interface Plan {
  id: string;
  name: string;
  code: string;
  priceWch: number;
  maxLiveBots: number;
  maxPaperBots: number;
  features: JsonObject;
  createdAt: Date;
}

export interface PlanWire {
  id: string;
  name: string;
  code: string;
  price_wch: number;
  max_live_bots: number;
  max_paper_bots: number;
  features: JsonObject;
  created_at: IsoDatetimeString;
}

export interface Subscription {
  id: string;
  userId: string;
  planId: string;
  status: SubscriptionStatus;
  paidAmountWch: number;
  startedAt: Date;
  expiresAt?: Date;
  createdAt: Date;
}

export interface SubscriptionWire {
  id: string;
  user_id: string;
  plan_id: string;
  status: SubscriptionStatus;
  paid_amount_wch: number;
  started_at: IsoDatetimeString;
  expires_at?: IsoDatetimeString;
  created_at: IsoDatetimeString;
}
